polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -935,6 +946,27 @@ impl<T: Config> Pallet<T> {
 		para: ParaId,
 		upward_messages: &[UpwardMessage],
 	) -> Result<(), UmpAcceptanceCheckErr> {
+		// Filter any pending UMP signals and the separator.
+		let upward_messages = if let Some(separator_index) =
+			upward_messages.iter().rposition(|message| message.is_empty())
```

Comment:
we still don't check here for multiple separators, as rposition will only find the last separator. I'd probably go for `position` instead

Reply:
changed to position.

Reply:
why is `position` better here? IIUC we still don't check for multiple separators

Reply:
We check that the length of ump signals together with a separator is 2. It is possible that the signal is a separator because we don't attempt to deserialize it.
Ideally, we'd have a single piece of logic that does validation and ump signal extraction and share it with `selected_core`. But this looks good enough already.